pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -89,19 +89,31 @@ func (cloud *CloudProvider) GetDefaultZone() string {
 
 // ListDisks lists disks based on maxEntries and pageToken only in the project
 // and zone that the driver is running in.
-func (cloud *CloudProvider) ListDisks(ctx context.Context, maxEntries int64, pageToken string) ([]*computev1.Disk, string, error) {
-	lCall := cloud.service.Disks.List(cloud.project, cloud.zone)
-	if maxEntries != 0 {
-		lCall = lCall.MaxResults(maxEntries)
-	}
-	if len(pageToken) != 0 {
-		lCall = lCall.PageToken(pageToken)
+func (cloud *CloudProvider) ListDisks(ctx context.Context) ([]*computev1.Disk, string, error) {
+	region, err := common.GetRegionFromZones([]string{cloud.zone})
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to get region from zones: %v", err)
 	}
-	diskList, err := lCall.Do()
+	zones, err := cloud.ListZones(ctx, region)
 	if err != nil {
 		return nil, "", err
 	}
-	return diskList.Items, diskList.NextPageToken, nil
+	items := []*computev1.Disk{}
+
+	for _, zone := range zones {
+		lCall := cloud.service.Disks.List(cloud.project, zone)
+		nextPageToken := "pageToken"
+		for nextPageToken != "" {
+			diskList, err := lCall.Do()
+			if err != nil {
+				return nil, "", err
+			}
+			items = append(items, diskList.Items...)
+			nextPageToken = diskList.NextPageToken
+			lCall.PageToken(nextPageToken)
+		}
```

Comment:
nit: this may be more concise and readable.	
```
    var nextToken string
	for {
		resp, err := cloud.service.Disks.List(cloud.project, zone).PageToken(nextToken).Do()
		if err != nil {
			// return error
		}
         items = append(items, diskList.Items...)
		if resp.NextPageToken == "" {
			break
		}
		nextToken = resp.NextPageToken
	}
```